src/sensor_lis2dw.c

```diff
@@ -6,32 +6,55 @@
 // This file may be distributed under the terms of the GNU GPLv3 license.
 
 #include <string.h> // memcpy
+#include "autoconf.h" // CONFIG_HAVE_GPIO_SPI
+#include "board/gpio.h" // irq_disable
 #include "board/irq.h" // irq_disable
 #include "board/misc.h" // timer_read_time
 #include "basecmd.h" // oid_alloc
 #include "command.h" // DECL_COMMAND
 #include "sched.h" // DECL_TASK
 #include "sensor_bulk.h" // sensor_bulk_report
-#include "spicmds.h" // spidev_transfer
+
+#if CONFIG_HAVE_GPIO_SPI
+ #include "spicmds.h" // spidev_transfer
+#endif
+
+#if CONFIG_HAVE_GPIO_I2C
+ #include "i2ccmds.h" // i2cdev_s
+#endif
```

Comment:
As mentioned previously, I think the ifdefs are too painful.  Just unconditionally require support for both.

I'm struggling to review the mcu code with all the ifdefs.  But, in general, I'd recommend avoiding the term "serial" and using DECL_ENUMERATION as mentioned earlier.

Reply:
I can change the naming, but as mentioned earlier I think this would be a regression in lis2dw support if we do not select these at build time…